src/providers/ogr/qgsogrdataitems.cpp

```diff
@@ -110,6 +114,38 @@ QString QgsOgrLayerItem::layerName() const
     return info.completeBaseName();
 }
 
+#ifdef HAVE_GUI
+QList<QAction *> QgsOgrLayerItem::actions()
+{
+  QList<QAction *> lst;
+  QAction *actionDeleteLayer = new QAction( tr( "Delete %1" ).arg( mName ), this );
+  connect( actionDeleteLayer, &QAction::triggered, this, &QgsOgrLayerItem::deleteLayer );
+  lst.append( actionDeleteLayer );
+  return lst;
+}
+
+void QgsOgrLayerItem::deleteLayer()
+{
+  if ( QMessageBox::question( nullptr, QObject::tr( "Delete Layer" ),
+                              QObject::tr( "Are you sure you want to delete %1?" ).arg( mName ),
+                              QMessageBox::Yes | QMessageBox::No, QMessageBox::No ) != QMessageBox::Yes )
+    return;
+
+  QString errCause;
+  bool res = ::deleteLayer( mUri, errCause );
+  if ( !res )
+  {
+    QMessageBox::warning( nullptr, tr( "Delete Layer" ), errCause );
+  }
+  else
+  {
+    QMessageBox::information( nullptr, tr( "Delete Layer" ), tr( "Layer deleted successfully." ) );
```

Comment:
For now, stick to the current implementation, like PostGIS and Spatialite already are doing: see

https://github.com/qgis/QGIS/blob/master/src/providers/postgres/qgspostgresdataitems.cpp#L345
https://github.com/qgis/QGIS/blob/master/src/providers/spatialite/qgsspatialitedataitems.cpp#L71

BTW, if we decide to change this behavior, it should be changed for all providers and I would say that it's out of scope for this PR,  we can do that in a separate PR.